jarvis/commands.py

```
@@ -121,8 +365,159 @@ def open_website(self, url):
+            data = urllib.parse.urlencode(
+                {"grant_type": "client_credentials"}
+            ).encode()
+            req = urllib.request.Request(
+                "https://accounts.spotify.com/api/token",
+                data=data,
+                headers={
+                    "Authorization": f"Basic {auth}",
+                    "Content-Type": "application/x-www-form-urlencoded",
+                },
+            )
+            with urllib.request.urlopen(req, timeout=5) as resp:
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)